api-conversion/src/main/java/io/strimzi/kafka/api/conversion/converter/MetricsConversion.java

```diff
@@ -0,0 +1,179 @@
+        }
+
+        String getKey() {
+            return get(".yaml");
+        }
+
+        String getName() {
+            return get("");
+        }
+
+        @Override
+        Class<T> convertedType() {
+            return clazz;
+        }
+
+        @Override
+        public InvertibleFunction<T> inverse() {
+            return new InverseMetricsInvertibleFunction<>(convertedType(), holderType);
+        }
+
+        @Override
+        public T apply(T holder) {
+            if (holder == null) {
+                return null;
+            }
+            Map<String, Object> metrics = holder.getMetrics();
+            if (metrics != null && holder.getMetricsConfig() == null) {
+                String key = getKey();
+                String name = getName();
+
+                JmxPrometheusExporterMetrics mc = new JmxPrometheusExporterMetrics();
+                ExternalConfigurationReference valueFrom = new ExternalConfigurationReference();
+                valueFrom.setConfigMapKeyRef(new ConfigMapKeySelector(key, name, null));
+                mc.setValueFrom(valueFrom);
+                holder.setMetricsConfig(mc);
+
+                StringWriter writer = new StringWriter();
+                try {
+                    MAPPER.writerFor(Map.class).with(new DefaultPrettyPrinter()).writeValue(writer, metrics);
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+                String metricsAsYaml = writer.toString();
+
+                ConfigMap configMap = new ConfigMapBuilder()
+                    .withMetadata(new ObjectMetaBuilder().withName(name).build())
+                    .withData(Collections.singletonMap(key, metricsAsYaml))
+                    .build();
+
+                MultipartConversions.get().addLast(new MultipartResource(
+                    name,
+                    configMap,
+                    (c, namespace) -> c.configMaps().inNamespace(namespace).createOrReplace(configMap)
+                ));
+
+                holder.setMetrics(null); // clear old metrics
+            }
+            return holder;
+        }
+    }
+
+    @Override
+    public Conversion<U> reverse() {
+        return reverse;
+    }
+
+    static class InverseMetricsInvertibleFunction<T extends HasConfigurableMetrics> extends MetricsInvertibleFunction<T> {
+        public InverseMetricsInvertibleFunction(Class<T> clazz, String holderType) {
+            super(clazz, holderType);
+        }
+
+        @Override
+        public InvertibleFunction<T> inverse() {
+            return new MetricsInvertibleFunction<>(convertedType(), holderType);
+        }
+
+        @Override
+        public T apply(T holder) {
+            if (holder == null) {
+                return null;
+            }
+
+            throw new UnsupportedOperationException("Reverse conversion of metrics is not supported");
```

Comment:
Not going to implement this?

Reply:
No, part of the _simplification_. I do not think this is really needed right now.